File: src/writePlan.ts
import { access, copyFile, writeFile } from "node:fs/promises";
import path from "node:path";
import { getWorkspaceCwd } from "./workspace.js";

const PLAN_FILENAME = "plan.md";
const PLAN_BAK_FILENAME = "plan.md.bak";

export async function writePlan(
  plan: string,
  cwd: string = getWorkspaceCwd(),
): Promise<string> {
  const planPath = path.join(cwd, PLAN_FILENAME);
  const bakPath = path.join(cwd, PLAN_BAK_FILENAME);
  try {
    await access(planPath);
    await copyFile(planPath, bakPath);
  } catch {
    // no existing plan.md
  }
  await writeFile(planPath, plan, "utf8");
  return planPath;
}

File: src/planner_ai/write_plan.py
from __future__ import annotations

import shutil
from pathlib import Path

from planner_ai.workspace import get_workspace_cwd

PLAN_FILENAME = "plan.md"
PLAN_BAK_FILENAME = "plan.md.bak"


def write_plan(plan: str, cwd: Path | None = None) -> Path:
    cwd = cwd or get_workspace_cwd()
    plan_path = cwd / PLAN_FILENAME
    bak_path = cwd / PLAN_BAK_FILENAME
    if plan_path.exists():
        shutil.copyfile(plan_path, bak_path)
    plan_path.write_text(plan, encoding="utf-8")
    return plan_path
